core/types/bal/bal_encoding.go

```diff
@@ -341,3 +342,31 @@ func (c *ContractCode) MarshalJSON() ([]byte, error) {
 	hexStr := fmt.Sprintf("%x", *c)
 	return json.Marshal(hexStr)
 }
+
+// UnmarshalJSON implements json.Unmarshaler to decode from RLP hex bytes
+func (b *BlockAccessList) UnmarshalJSON(input []byte) error {
+	// Handle both hex string and object formats
+	var hexBytes hexutil.Bytes
+	if err := json.Unmarshal(input, &hexBytes); err == nil {
+		// It's a hex string, decode from RLP
+		return rlp.DecodeBytes(hexBytes, b)
+	}
+	
+	// Otherwise try to unmarshal as structured JSON
+	var tmp []AccountAccess
+	if err := json.Unmarshal(input, &tmp); err != nil {
+		return err
+	}
+	*b = BlockAccessList(tmp)
+	return nil
+}
+
+// MarshalJSON implements json.Marshaler to encode as RLP hex bytes
+func (b BlockAccessList) MarshalJSON() ([]byte, error) {
```

Comment:
I already merged these commits.  But realized that we definitely don't want the JSON marshaller defaulting to encode the BAL as a hex string.